#Represents if we are done approaching
bool doneWithState
